python/ray/util/state/common.py

```diff
@@ -24,16 +24,25 @@
 from ray.util.state.exception import RayStateApiException
 
 try:
+    import pydantic
     from pydantic.dataclasses import dataclass
 
     from ray.dashboard.modules.job.pydantic_models import JobDetails
 
+    # TODO(aguo): Instead of a version check, modify these classes to use
+    # pydantic BaseModel instead of dataclass.
+    # In pydantic 2, dataclass no longer needs the `init=True` kwarg to
+    # generate an __init__ method. Additionally, it will raise an error if
+    # it detects `init=True` to be set.
+    is_pydantic_2 = pydantic.__version__.startswith("2")
+
 except ImportError:
     # pydantic is not available in the dashboard.
     # We will use the dataclass from the standard library.
     from dataclasses import dataclass
 
     JobDetails = object
+    is_pydantic_2 = False
```

Comment:
why don't we just call it `pydantic_1_available`? This seems to be more clear with code below

Reply:
that naming doesn't quite work, because when pydantic is not installed at all, also want `init=True`. Setting the variable `pydantic_1_available = True` is weird.